e2e-tests/portal/cypress/e2e/journeys/maker-about-supply-contract/fill-in-about-supply-contract-validation.spec.js

```diff
@@ -160,7 +160,7 @@ context('about-supply-contract', () => {
     cy.url().should('eq', relative(`/contract/${dealId}/about/supplier`));
 
     // should only see companies house validation errors
-    partials.errorSummary.errorSummaryLinks().should('have.length', 1);
+    partials.errorSummary.errorSummaryLinks().should('have.length', 12);
```

Comment:
This shouldnt be changed like this -- the test is checking only that companies house validation errors are present. If there are more than this one error it means that either our test is wrong or our code is wrong

Reply:
When we entered the companies and come back second time the same page. Still we need to show the error message. So we update the code, that is reason we update the e2e tests also.

Reply:
The comment says `    // should only see companies house validation errors` -- It feels that as a user if I search for a companies house number, I would not expect the rest of the page to error at unfilled in data (especially as it's at the top of the page!). @abhi-markan do you know about any intended behaviour in this regard or should Vamshi contact the ui/ux guys please?

Reply:
When the user searches companies house number with invalid data or empty filed just the companies house validation error is displayed. If valid companies house number is entered that values are populated for rest of of the fields except for supplier type, supplier correspondence address, industry sector, industry class, sme type, description, guarantor and if user clicks on Next and navigates back validation errors displayed only for the missing fields.

Reply:
Had a call with Abhi on this. Abhi confirmed that the error displayed is as expected.

Reply:
I checked the test the companies house validation error is added to the list of errors on supplier page so ether the comment needs to be removed or should we need to come up with new test.

Reply:
I came up with below code and the test pass which checks length of 1 for companies house. Please let me know if this is correct approach to test companies house validation error.                                                                          ```
it('trigger just Supplier companies house validation error', () => {
    cy.login(BANK1_MAKER1);
    // dashboard.createNewSubmission().click();
    cy.visit('/dashboard');
    // Select "Create new" and choose "Bond Support" as the scheme type
    cy.get('[data-cy="CreateNewSubmission"]').click();
    cy.get('[data-cy="scheme-bss"]').click();
    cy.get('[data-cy="continue-button"]').click();
    cy.get('[data-cy="criteriaMet-true"]').click();
    cy.get('[data-cy="submit-button"]').click();
    cy.get('[data-cy="bankInternalRefName"]').type('TestBank1903');
    cy.get('[data-cy="additionalRefName"]').type('TestBank1903');
    cy.get('[data-cy="submit-button"]').click();

    // Click on "View Details" under the "About the Supplier" contract
    cy.get('[data-cy="ViewAboutSupplierDetails"]').click();
    // Enter an invalid Companies House registration number
    contractAboutSupplier.supplierCompaniesHouseRegistrationNumber().type('TEST');

    // Click on Search
    contractAboutSupplier.supplierSearchCompaniesHouse().click();

    // Check that the correct validation error is displayed
    partials.errorSummary.errorSummaryLinks().should('have.length', 1);
    contractAboutSupplier.expectError('Enter a valid Companies House registration number');
    // Enter an empty Companies House registration number
    contractAboutSupplier.supplierCompaniesHouseRegistrationNumber().clear();
    contractAboutSupplier.supplierSearchCompaniesHouse().click();
    // Check that the correct validation error is displayed
    partials.errorSummary.errorSummaryLinks().should('have.length', 1);
    contractAboutSupplier.expectError('Enter a valid Companies House registration number'); 
});
```

Reply:
Added test to check just companies house validation error